scipy/special/tests/test_data.py

```diff
@@ -562,7 +562,9 @@ def clog1p(x, y):
 
 @pytest.mark.parametrize('test', BOOST_TESTS, ids=repr)
 def test_boost(test):
-    _test_factory(test)
+    with suppress_warnings() as sup:
+        sup.filter(DeprecationWarning)
+        _test_factory(test)
```

Comment:
Would prefer to use the new name rather than filter the warnings

Reply:
In the recent update, I restored `btdtr` and `bdtri` to their original wrappers of the Cephes functions, and deprecated them.  The tests for these functions should still be run.

We *could* make the filter more specific, so the deprecation warning is only filtered for `btdtr` and `btdtri`.  I guess the point of such a change would be to avoid accidentally filtering an incorrect or unexpected `DeprecationWarning`?

Reply:
This should be as easy as replacing `"btdtr"` -> `"betainc"`, `"btdtri"` -> `"betaincinv"` in the definition of `BOOST_TESTS` above. Then you can remove the warning filter again.

Reply:
`betainc` and `betaincinv` are already in `BOOST_TESTS`.  As I noted earlier, `btdtr` and `btdri` are no longer aliases of `betainc` and `betaincinv`; the only changes made to `btdtr` and `btdtri` are the deprecation.  Since they are using different code than that of `betainc` and `betaincinv`, their tests should still be run.  When `btdtr` and `btdtri` are removed at the end of their deprecation period, these tests can be removed.

Answering my own question above, I think it is better to only filter expected deprecation warnings, so updated the test accordingly.